crates/noirc_frontend/src/monomorphization/mod.rs

```diff
@@ -755,13 +774,16 @@ impl<'interner> Monomorphizer<'interner> {
     ) -> Option<ast::Expression> {
         if let ast::Expression::Ident(ident) = func {
             if let Definition::Builtin(opcode) = &ident.definition {
+                // TODO(#1736): Move this builtin to the SSA pass
                 if opcode == "array_len" {
                     let typ = self.interner.id_type(arguments[0]);
-                    let len = typ.evaluate_to_u64().unwrap();
-                    return Some(ast::Expression::Literal(ast::Literal::Integer(
-                        (len as u128).into(),
-                        ast::Type::Field,
-                    )));
+                    if let Type::Array(_, _) = typ {
+                        let len = typ.evaluate_to_u64().unwrap();
```

Comment:
Instead of checking if it is an array, does `if let Some(len) = typ.evaluate_to_u64()` work?

Reply:
Hmmm I think so so I changed it. `evaluate_to_u64` is only used for Type::Constant or types bound to a Type::Constant either way and this check will be removed when we move to the new SSA

Reply:
Ah actually this causes us to hit an unwrap error as we may have a slice. I kept the switch but I added the `enable_slices` flag check

Reply:
Ah, you can keep it how it was before then. I'd rather minimize uses of `enable_slices`.